repo/fsrepo/migrations/ipfsfetcher/ipfsfetcher.go

```diff
@@ -240,21 +231,22 @@ func (f *IpfsFetcher) startTempNode(peers []peer.AddrInfo) error {
 		fmt.Println("migration peer", node.Identity, "shutdown")
 	}
 
-	// Parse peer addresses and asynchronously connect to peers
-	if len(peers) != 0 {
-		connectPeers(ctxIpfsLife, ifaceCore, peers)
+	// Asynchronously connect to peers
+	if len(f.peers) != 0 {
+		go connectPeers(ctxIpfsLife, ipfs, f.peers)
 	}
 
-	a, err := ma.NewMultiaddr(tempNodeTcpAddr)
+	addrs, err := ipfs.Swarm().LocalAddrs(ctx)
```

Comment:
What's the idea behind where the background context is getting used and where the passed in context is getting used?

If this function doesn't complete then the temporary node is useless right (aside from the fact that if MDNS is working then everything is probably fine)?

If we don't get the addresses do we need to error + close the node?

Reply:
The background context is used as a mechanism to shutdown the temporary node. Quote from `go-ipfs/core/builder.go`:
>      // Shut down the application if the lifetime context is canceled.
>      // NOTE: we _should_ stop the application by calling `Close()` 
>      // on the process. But we currently manage everything with contexts.                                   

So, the background context is for now shutdown, in the absence of a `node.Close()` method, and the passed-in context is used for cancellation of the current function call.  

True, if the function does not complete, then the node is useless.  However, failure to get the local swarm address only means that the downloaded migrations cannot be fetched from the temp node.... So, right, it does not need to fail out here.